src/internal/connector/onedrive/collections_test.go

```diff
@@ -969,7 +972,419 @@ func (suite *OneDriveCollectionsSuite) TestDeserializeMetadata() {
 	}
 }
 
-func driveItem(id string, name string, parentPath string, isFile, isFolder, isPackage bool) models.DriveItemable {
+type mockDeltaPageLinker struct {
+	link  *string
+	delta *string
+}
+
+func (pl *mockDeltaPageLinker) GetOdataNextLink() *string {
+	return pl.link
+}
+
+func (pl *mockDeltaPageLinker) GetOdataDeltaLink() *string {
+	return pl.delta
+}
+
+type deltaPagerResult struct {
+	items     []models.DriveItemable
+	nextLink  *string
+	deltaLink *string
+	err       error
+}
+
+type mockItemPager struct {
+	// DriveID -> set of return values for queries for that drive.
+	toReturn []deltaPagerResult
+	getIdx   int
+}
+
+func (p *mockItemPager) GetPage(context.Context) (gapi.DeltaPageLinker, error) {
+	if len(p.toReturn) <= p.getIdx {
+		return nil, assert.AnError
+	}
+
+	idx := p.getIdx
+	p.getIdx++
```

Comment:
oh that's a good point. If it's alright for you I'd like to leave things as they are even if it's not as elegant. What I would like to avoid at the moment is having to start the index at a non-zero value for new instances. If we move mocks to a different package then I think it would be a good addition. However, with golang's current export rules and having the mock code in the same package I'm not certain of the ability to keep the constructor as the only initializer used to create mock instances. That could possibly lead to test failures (not too bad but can be annoying) or tests that seem to work but actually don't (much harder to find and debug)

If you feel strongly about it I can make a follow-up PR to patch the code here and maybe in another place the same pattern is used